pkg/collector/collector.go

```diff
@@ -22,6 +22,8 @@ const (
 	started
 )
 
+const cancelCheckTimeout time.Duration = 500 * time.Millisecond
```

Comment:
should we make it configurable ?

Reply:
I think the `Cancel` method should be implemented to return immediately, i.e: set a flag/trigger a cleanup and return (I can make that clearer in the inline docs). So this timeout is only there as a safeguard to not block the entire collector. WDYT?

Reply:
agreed